### DISCUSSION TITLE
Python-Rust Interaction

### DISCUSSION THREAD

**Discussion by Moosems** [astral-sh/ruff#4273] [General]
How does Python talk to Rust in this case and how could this be applied in other projects?

**Comment by konstin** [comment]
ruff itself doesn't use cpython or any python interpreter at all. Instead, we compile a regular binary with cargo and ship that inside of a wheel ([maturin](https://github.com/PyO3/maturin) does the wrapping part). On unix, the binary will be installed to `bin/ruff` side your virtualenv, on windows to `Scripts\ruff.exe`; It's just a normal binary like other applications.

If we were to make ruff callable from python, we would likely add [pyo3](https://github.com/PyO3/pyo3) bindings.
